src/python/pants/backend/jvm/tasks/coursier_resolve.py

```diff
@@ -0,0 +1,583 @@
+
+        results = self._get_result_from_coursier(global_excludes,
+                                                 jars_to_resolve, pants_workdir,
+                                                 pinned_coords, coursier_cache_path)
+
+        for classifier, result in results.items():
+          self._load_json_result(classifier, compile_classpath, coursier_cache_path, invalidation_check,
+                                 pants_jar_path_base, result)
+
+        self._populate_results_dir(vt_set_results_dir, results)
+        resolve_vts.update()
+
+  def _get_result_from_coursier(self, global_excludes, jars_to_resolve,
+                                pants_workdir, pinned_coords, coursier_cache_path):
+    """
+    Calling coursier and return the result per invocation.
+
+    If coursier was called once for classifier '' and once for classifier 'tests', then the return value
+    would be: {'': <first couriser output>, 'tests': <second coursier output>}
+
+    :param global_excludes:
+    :param jars_to_resolve:
+    :param local_exclude_args:
+    :param pants_workdir:
+    :param pinned_coords:
+    :param coursier_cache_path:
+    :return:
+    """
+
+    # Prepare coursier args
+    coursier_subsystem_instance = CoursierSubsystem.global_instance()
+    coursier_jar = coursier_subsystem_instance.bootstrap_coursier(self.context.new_workunit)
+
+    coursier_workdir = os.path.join(pants_workdir, 'tmp')
+    safe_mkdir(coursier_workdir)
+    with temporary_file(coursier_workdir, cleanup=False) as f:
+      output_fn = f.name
+    common_args = ['fetch',
+                   # Print the resolution tree
+                   '-t',
+                   '--cache', coursier_cache_path,
+                   '--json-output-file', output_fn] + coursier_subsystem_instance.get_options().fetch_options
+
+    def construct_classifier_to_jar(jars):
+      """
+      Reshape jars by classifier
+
+      For example:
+      [
+        ResolvedJar('sources', 'a-sources.jar'),
+        ResolvedJar('sources', 'b-sources.jar'),
+        ResolvedJar('', 'a.jar'),
+      ]
+      should yield:
+      {
+        'sources': [ResolvedJar('sources', 'a-sources.jar'),ResolvedJar('sources', 'b-sources.jar')],
+        '':  [ResolvedJar('', 'a.jar') ]
+      }
+      """
+      product = defaultdict(list)
+      for x in jars:
+        product[x.coordinate.classifier or ''].append(x)
+      return product
+
+    classifier_to_jars = construct_classifier_to_jar(jars_to_resolve)
+    # Divide the resolve by classifier because coursier treats classifier option globally.
+
+    results = {}
+
+    for classifier, classified_jars in classifier_to_jars.items():
+
+      cmd_args = self._construct_cmd_args(classified_jars, classifier, common_args, global_excludes,
+                                          pinned_coords, coursier_workdir)
+
+      cmd_str = ' '.join(cmd_args)
+      logger.debug(cmd_str)
```

Comment:
This should use `self.context.log` and `import logging` should be dropped. But it would already be logged by `runjava` I believe.